diff --git a/src/alejandra/src/rules/list.rs b/src/alejandra/src/rules/list.rs
index 96a7690..d725015 100644
--- a/src/alejandra/src/rules/list.rs
+++ b/src/alejandra/src/rules/list.rs
@@ -23,7 +23,7 @@ pub(crate) fn rule(
     steps.push(crate::builder::Step::Format(child));
     if vertical {
         steps.push(crate::builder::Step::Indent);
-    } else if items_count > 0 && build_ctx.config.space_around_brackets {
+    } else {
         steps.push(crate::builder::Step::Whitespace);
     }

@@ -92,7 +92,7 @@ pub(crate) fn rule(
         steps.push(crate::builder::Step::Dedent);
         steps.push(crate::builder::Step::NewLine);
         steps.push(crate::builder::Step::Pad);
-    } else if items_count > 0 && build_ctx.config.space_around_brackets {
+    } else if items_count > 0 {
         steps.push(crate::builder::Step::Whitespace);
     }
     steps.push(crate::builder::Step::Format(child));
